pandas/core/strings/accessor.py

```diff
@@ -3033,16 +3033,26 @@ def cat_core(list_of_columns: List, sep: str):
 
 def _groups_or_na_fun(regex):
     """Used in both extract_noexpand and extract_frame"""
-    empty_row = [np.nan] * regex.groups
-
-    def f(x):
-        if not isinstance(x, str):
-            return empty_row
-        m = regex.search(x)
-        if m:
-            return [np.nan if item is None else item for item in m.groups()]
-        else:
-            return empty_row
+
+    if regex.groups == 1:
+
+        def f(x):
+            if not isinstance(x, str):
+                return np.nan
```

Comment:
once we dispatch to array we can remove this isinstance check but will probably only make a small difference, maybe 2-3%